gtsam/hybrid/tests/testHybridIncremental.cpp

```diff
@@ -150,497 +143,487 @@ TEST(HybridGaussianElimination, IncrementalInference) {
-//   // 0 has 2 choices, 1 has 4 choices and 2 has 4 choices.
-//   inc.update(gfg, ordering, 2);
-
-//   /*************** Run Round 4 ***************/
-//   // Add odometry factor with discrete modes.
-//   contKeys = {W(2), W(3)};
-//   still = boost::make_shared<PlanarMotionModel>(W(2), W(3), Pose2(0, 0, 0),
-//                                                 noise_model);
-//   moving =
-//       boost::make_shared<PlanarMotionModel>(W(2), W(3), odometry,
-//       noise_model);
-//   components = {moving, still};
-//   dcFactor = boost::make_shared<DCMixtureFactor<PlanarMotionModel>>(
-//       contKeys, DiscreteKeys{gtsam::DiscreteKey(M(3), 2)}, components);
-//   fg.push_back(dcFactor);
-
-//   // Add equivalent of ImuFactor
-//   fg.emplace_nonlinear<BetweenFactor<Pose2>>(X(2), X(3), Pose2(1.0, 0.0, 0),
-//                                              poseNoise);
-//   // PoseFactors-like at k=3
-//   fg.emplace_nonlinear<BetweenFactor<Pose2>>(X(3), Y(3), Pose2(0, 1, 0),
-//                                              poseNoise);
-//   fg.emplace_nonlinear<BetweenFactor<Pose2>>(Y(3), Z(3), Pose2(0, 1, 0),
-//                                              poseNoise);
-//   fg.emplace_nonlinear<BetweenFactor<Pose2>>(Z(3), W(3), Pose2(-3, 1, 0),
-//                                              poseNoise);
-
-//   initial.insert(X(3), Pose2(3.0, 0.0, 0.0));
-//   initial.insert(Y(3), Pose2(3.0, 1.0, 0.0));
-//   initial.insert(Z(3), Pose2(3.0, 2.0, 0.0));
-//   initial.insert(W(3), Pose2(0.0, 3.0, 0.0));
-
-//   // Ordering at k=3. Same intuition as before.
-//   ordering = Ordering();
-//   ordering += W(2);
-//   ordering += Z(2);
-//   ordering += Y(2);
-//   ordering += X(2);
-//   ordering += W(3);
-//   ordering += Z(3);
-//   ordering += Y(3);
-//   ordering += X(3);
-
-//   gfg = fg.linearize(initial);
-//   fg = NonlinearHybridFactorGraph();
-
-//   // Keep pruning!
-//   inc.update(gfg, ordering, 3);
-
-//   // The final discrete graph should not be empty since we have eliminated
-//   // all continuous variables.
-//   EXPECT(!inc.remainingDiscreteGraph().empty());
-
-//   // Test if the optimal discrete mode assignment is (1, 1, 1).
-//   DiscreteValues optimal_assignment =
-//   inc.remainingDiscreteGraph().optimize(); DiscreteValues
-//   expected_assignment; expected_assignment[M(1)] = 1;
-//   expected_assignment[M(2)] = 1;
-//   expected_assignment[M(3)] = 1;
-//   EXPECT(assert_equal(expected_assignment, optimal_assignment));
-
-//   // Test if pruning worked correctly by checking that we only have 3 leaves
-//   in
-//   // the last node.
-//   auto lastConditional = boost::dynamic_pointer_cast<GaussianMixture>(
-//       inc.hybridBayesNet().at(inc.hybridBayesNet().size() - 1));
-//   EXPECT_LONGS_EQUAL(3, lastConditional->nrComponents());
-// }
+// Test for figuring out the optimal ordering to ensure we get
+// a discrete graph after elimination.
+TEST(HybridGaussianISAM, NonTrivial) {
```

Comment:
Is this one meant to be commented out?

Reply:
Yeah I am working on getting this to pass. I might need to open a separate PR though since it has some non-trivial changes.